crates/sui-core/src/authority_active/gossip/mod.rs

```diff
@@ -239,13 +328,21 @@ where
         }
     }
 
-    pub async fn start(mut self, duration: Duration) -> (AuthorityName, Result<(), SuiError>) {
+    pub async fn start<Handler: FollowerDigestHandler<A>>(
+        self,
+        duration: Duration,
+        handler: Handler,
+    ) -> (AuthorityName, Result<(), SuiError>) {
         let peer_name = self.peer_name;
-        let result = self.peer_gossip_for_duration(duration).await;
+        let result = self.follow_peer_for_duration(duration, handler).await;
```

Comment:
In the case of gossip we do not want to follow from the last known seq, but in the case of sync we do. How do we distinguish between the two when we start this process?

Reply:
We should have that be a config variable, along with all the other hardcoded values.

Reply:
I think it doesn't have to be config driven, since we have a different entry point for gossip and sync. I will try to straighten this out soon unless you want to take care of it, @laura-makdah

Reply:
Yes, feel free to merge this in. I also like incremental PRs.